taskmgr/queueservice.go

```diff
@@ -53,22 +44,34 @@ type (
 		TaskClaimResponse   tcqueue.TaskClaimResponse      `json:"-"`
 		TaskReclaimResponse tcqueue.TaskReclaimResponse    `json:"-"`
 		Definition          tcqueue.TaskDefinitionResponse `json:"-"`
-		reclaimTimer        *time.Timer
 	}
 
-	TaskQueue struct {
+	taskQueue struct {
 		SignedDeleteUrl string `json:"signedDeleteUrl"`
 		SignedPollUrl   string `json:"signedPollUrl"`
 	}
 
-	TaskRuns []*TaskRun
+	taskRuns []*TaskRun
+
+	// TaskclusterQueue is an interface to the Queue client provided by the
+	// taskcluster-client-go package.  Passing around an interface allows the
+	// queue client to be mocked
+	TaskclusterQueue interface {
+		ReportCompleted(string, string) (*tcqueue.TaskStatusResponse, *tcclient.CallSummary, error)
+		ReportException(string, string, *tcqueue.TaskExceptionRequest) (*tcqueue.TaskStatusResponse, *tcclient.CallSummary, error)
+		ReportFailed(string, string) (*tcqueue.TaskStatusResponse, *tcclient.CallSummary, error)
+		ClaimTask(string, string, *tcqueue.TaskClaimRequest) (*tcqueue.TaskClaimResponse, *tcclient.CallSummary, error)
+		ReclaimTask(string, string) (*tcqueue.TaskReclaimResponse, *tcclient.CallSummary, error)
+		PollTaskUrls(string, string) (*tcqueue.PollTaskUrlsResponse, *tcclient.CallSummary, error)
+		CancelTask(string) (*tcqueue.TaskStatusResponse, *tcclient.CallSummary, error)
+	}
```

Comment:
Here it would be nice to declare names for the parameters (e.g. `CencelTask(taskId string)`) to make it clearer what each parameter is for.